File: Binary Search/binary-search.js
const binarySearch =(arr, item) =>{
    let low = 0;
    let high = arr.length - 1;
    while(low <= high){
        guess = Math.floor((low + high ) / 2);
        if(item == arr[guess]) 
            return guess;
        else if (item < arr[guess])
            high = guess - 1;
        else 
            low = guess + 1;
    }
    return null; 
};

console.log(binarySearch([1, 3, 5, 7, 9], 7))
console.log(binarySearch([1, 3, 5, 7, 9], 8))
console.log(binarySearch([1, 3, 5, 7, 9], 3))
console.log(binarySearch([1, 3, 5, 7, 9], 9))

File: Binary Search/binary-search.py
# Binary Search Python Code
def binary_search(arr, item):
    low = 0
    high = len(arr) - 1
    while low <= high:
        guess = (low + high) // 2
        if arr[guess] == item:
            return guess
        elif arr[guess] > item:
            high = guess - 1
        else:
            low = guess + 1
    return None



print(binary_search([1, 3, 5, 7, 9], 7))
print(binary_search([1, 3, 5, 7, 9], 8))
print(binary_search([1, 3, 5, 7, 9], 3))
print(binary_search([1, 3, 5, 7, 9], 9))

